diff -rup -x*.1 -x*.exe -x*.o -x*.a -x*.dll -x*CVS* diffutils-2.8.1-orig/lib/c-stack.c diffutils-2.8.1-mingw/lib/c-stack.c
--- diffutils-2.8.1-orig/lib/c-stack.c	Sun Mar 24 07:36:38 2002
+++ diffutils-2.8.1-mingw/lib/c-stack.c	Sat Jan  4 18:57:54 2003
@@ -156,7 +156,9 @@ c_stack_die (int signo, siginfo_t *info,
       *addr = 0;
     }
 #endif
+#if HAVE_WORKING_FORK || HAVE_WORKING_VFORK
   kill (getpid (), signo);
+#endif
 }
 
 
diff -rup -x*.1 -x*.exe -x*.o -x*.a -x*.dll -x*CVS* diffutils-2.8.1-orig/lib/tempname.c diffutils-2.8.1-mingw/lib/tempname.c
--- diffutils-2.8.1-orig/lib/tempname.c	Wed Feb  6 05:36:12 2002
+++ diffutils-2.8.1-mingw/lib/tempname.c	Sat Jan  4 18:57:54 2003
@@ -90,6 +90,10 @@
 # define S_IXUSR 00100
 #endif
 
+#ifdef _WIN32
+# define mkdir(foo, bar) mkdir(foo)
+#endif
+
 #if _LIBC
 # define struct_stat64 struct stat64
 #else
diff -rup -x*.1 -x*.exe -x*.o -x*.a -x*.dll -x*CVS* diffutils-2.8.1-orig/lib/waitpid.c diffutils-2.8.1-mingw/lib/waitpid.c
--- diffutils-2.8.1-orig/lib/waitpid.c	Fri Mar  1 07:46:32 2002
+++ diffutils-2.8.1-mingw/lib/waitpid.c	Sat Jan  4 19:04:54 2003
@@ -25,6 +25,10 @@
 extern int errno;
 #endif
 
+#ifdef __MINGW32__
+#include <sys/types.h>
+#endif
+
 #define WAITPID_CHILDREN 8
 static pid_t waited_pid[WAITPID_CHILDREN];
 static int waited_status[WAITPID_CHILDREN];
diff -rup -x*.1 -x*.exe -x*.o -x*.a -x*.dll -x*CVS* diffutils-2.8.1-orig/src/sdiff.c diffutils-2.8.1-mingw/src/sdiff.c
--- diffutils-2.8.1-orig/src/sdiff.c	Mon Mar 11 08:00:48 2002
+++ diffutils-2.8.1-mingw/src/sdiff.c	Sat Jan  4 19:09:10 2003
@@ -83,8 +83,10 @@ static int const sigs[] = {
 #ifdef SIGXFSZ
        SIGXFSZ,
 #endif
-       SIGINT,
-       SIGPIPE
+       SIGINT
+#ifdef SIGPIPE
+       ,SIGPIPE
+#endif
 };
 #define handler_index_of_SIGINT (NUM_SIGS - 2)
 #define handler_index_of_SIGPIPE (NUM_SIGS - 1)
@@ -838,8 +840,10 @@ checksigs (void)
       cleanup ();
 
       /* Yield an exit status indicating that a signal was received.  */
+#if HAVE_WORKING_FORK || HAVE_WORKING_VFORK
       untrapsig (s);
       kill (getpid (), s);
+#endif
 
       /* That didn't work, so exit with error status.  */
       exit (EXIT_TROUBLE);
@@ -1193,6 +1197,13 @@ diraccess (char const *dir)
 # define TMPDIR_ENV "TMPDIR"
 #endif
 
+/* FIXME: mkstemp is declared in src/mkstemp.c */
+#ifdef __MINGW32__
+ #ifndef mkstemp
+  #define mkstemp(foo) (-1)
+ #endif
+#endif
+
 /* Open a temporary file and return its file descriptor.  Put into
    tmpname the address of a newly allocated buffer that holds the
    file's name.  Use the prefix "sdiff".  */
@@ -1207,14 +1218,18 @@ temporary_file (void)
   sigset_t procmask;
   sigset_t blocked;
   sprintf (buf, "%s/sdiffXXXXXX", dir);
+#if HAVE_WORKING_VFORK
   sigemptyset (&blocked);
   sigaddset (&blocked, SIGINT);
   sigprocmask (SIG_BLOCK, &blocked, &procmask);
+#endif
   fd = mkstemp (buf);
   e = errno;
   if (0 <= fd)
     tmpname = buf;
+#if HAVE_WORKING_VFORK
   sigprocmask (SIG_SETMASK, &procmask, 0);
+#endif
   errno = e;
   return fd;
 }
